core/units.c

```diff
@@ -183,3 +186,49 @@ const struct units *get_units()
 {
 	return &prefs.units;
 }
+
+int parse_duration(const char *buffer, duration_t *time, bool dot_as_seperator)
+{
+	char *line = (char*)buffer;
+	size_t n = 0;
+	long values[3];
+	char separator;
+	while (n < sizeof(values)/sizeof(values[0])) {
+		values[n++] = strtol(line, &line, 10);
+		separator = *line;
+		line++;
+		/* special case of `mm.ss`
+		 * DivingLog 5.08 (and maybe other versions) appear to sometimes
+		 * store the dive time as 44.00 instead of 44:00 */
+		if (dot_as_seperator && n == 1 && separator == '.')
+			continue;
+		if (separator != ':')
+			break;
+	}
+	switch (n) {
+	case 1:
+		/* mm */
+		time->seconds = values[0] * 60;
+		time->ms = 0;
+		goto exit_out;
```

Comment:
I don't understand the control-flow here.

This seems to be considering the case `11.111` when `dot_as_separator` is `false`. Ok, but the code above skipped over the `.`. Is this what we want (i.e. consume the `.`)?